### PR TITLE
groups: rework of group creation flow

### PR DESCRIPTION
**Pull request by rcrdlbl**
see: #788.

there are still a few things I may have missed with this w/r/t styling, but everything seems to work well.

I still wasn't able to figure out what was causing #651, however.

### CODE REVIEW

File: ui/src/groups/GroupInfoFields.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React, { useEffect, useState } from 'react';
 import { useFormContext } from 'react-hook-form';
 import ColorBoxIcon from '@/components/icons/ColorBoxIcon';
```
Comment (patosullivan) [inline ui/src/groups/GroupInfoFields.tsx:3]: We can probably ditch all of these unused imports.

File: ui/src/groups/NewGroup/GroupInfoPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import cn from 'classnames';
+import { useFormContext } from 'react-hook-form';
```
Comment (patosullivan) [inline ui/src/groups/NewGroup/GroupInfoPreview.tsx:3]: Same here, we can get rid of these unused imports.

**Review by patosullivan — APPROVED** [review]
Looks good, one nit: we should probably get rid of the unused imports. I noticed them in two of these files, but there may be others.

**Review by urcades — APPROVED** [review]
@rcrdlbl this is looking good, the initial state of the creation flow is beautiful and implemented exactly as intended.

A minor bit of feedback that's polish-oriented (and would be nice to have, but shouldn't block merge) is to potentially "fix"/"pin" the top edge of the modal so that it's not jumping around vertically as one clicks through the options, as shown in the video below:

https://user-images.githubusercontent.com/1195363/199289084-fe028426-e6c2-49a6-9505-bae73f4f7354.mov

**Comment by rcrdlbl** [comment]
@urcades I've updated this – note that this approach inevitably makes it much more difficult to keep the modal exactly centered on the screen

**Review by jamesacklin — APPROVED** [review]

**Pull request merged**